src/recordlinker/linkage/models.py

```diff
@@ -47,3 +45,46 @@ class BlockingValue(Base):
     patient_id: orm.Mapped[int] = orm.mapped_column(ForeignKey("mpi_patient.id"))
     blockingkey_id: orm.Mapped[int] = orm.mapped_column(ForeignKey("mpi_blocking_key.id"))
     value: orm.Mapped[str] = orm.mapped_column(String(50), index=True)
+
+class Algorithm(Base):
+    __tablename__ = "algorithm"
+
+    id: orm.Mapped[int] = orm.mapped_column(primary_key=True)
+    is_default: orm.Mapped[bool] = orm.mapped_column(default=False, index=True)
+    label: orm.Mapped[str] = orm.mapped_column(String(255), unique=True)
+    description: orm.Mapped[str] = orm.mapped_column(Text())
+
+def check_only_one_default(mapping, connection, target):
+    """
+    Check if there is already a default algorithm before inserting or updating.
+    If another default algorithm exists, an exception is raised to prevent the operation.
+
+    Parameters:
+    connection: The database connection being used for the operation.
+    target: The instance of the Algorithm class being inserted or updated.
+    
+    Raises:
+    ValueError: If another algorithm is already marked as default.
+    """
+
+    session = orm.Session.object_session(target)
+  
+    if target.is_default:
+        existing = session.query(Algorithm).filter(Algorithm.is_default == True).first()    # noqa
```

Comment:
@cbrinson-rise8 your comment on why this should be ignored is really helpful, mind putting that in the code for future eyes?  Also if we can just ignore that 1 rule, that be great.
```suggestion
        existing = session.query(Algorithm).filter(Algorithm.is_default == True).first()    # noqa: E712
```

Reply:
added comment explanation and the specific rule number